### PR TITLE
Recommend the Clang-14 tarball

### PR DESCRIPTION
**Pull request by djmitche**
Newer versions of this Clang+LLVM budnel are not available in the Clang releases, and other means of installing Clang-16 suggest that it fails building various Envoy dependencies. Clang-14 is used in CI, so recommend that version.

This is based on my experience trying to build Envoy from source (without using the docker image).

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -98,7 +98,9 @@ for how to update or override dependencies.
     ```
 
     ### Linux
-    On Linux, we recommend using the prebuilt Clang+LLVM package from [LLVM official site](http://releases.llvm.org/download.html).
+    On Linux, we recommend using the prebuilt Clang+LLVM package from [LLVM official site](http://releases.llvm.org/download.html) for Clang 14.
+    Note that newer versions of Clang do not include this package for Linux, but neither are newer versions of Clang able to build the Envoy dependencies.
```
Comment (phlax) [inline bazel/README.md:102]: newer versions dont work for a bunch of reasons - there are some tarballs available - but the codebase needs to be updated before we can use them.

I would suggest just removing this bit

**Review by phlax — COMMENTED** [review]
hi @djmitche thanks for raising this

atm we are stuck on llvm 14 - i have made various attempts at upgrade to 15 and 16, but both require other changes, in particular other deps to be updated.

fwiw the recommended way to build is using the envoy build container (ie `./ci/run_envoy_docker.sh` as this guarantees correct versions

File: bazel/README.md (hunk 1 above)
Comment (djmitche) [reply bazel/README.md]: Done - thanks!

**Review by phlax — APPROVED** [review]
lgtm, thanks @djmitche

**Pull request merged**